react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = ({ currentIndex: 0 });
+    this.renderBalls = this.renderBalls.bind(this);
+    this.changeImage = this.changeImage.bind(this);
+    this.chevronClick = this.chevronClick.bind(this);
+    this.goToBall = this.goToBall.bind(this);
+    this.intervalID = setInterval(this.changeImage, 3000);
```

Comment:
You should do this in `componentDidMount()`

https://reactjs.org/docs/react-component.html#componentdidmount